models/lobby/server.go

```diff
@@ -0,0 +1,255 @@
+package lobby
+
+import (
+	"fmt"
+	"log"
+	"strconv"
+	"time"
+
+	"github.com/TeamPlayTF/TF2RconWrapper"
+)
+
+type Server struct {
+	Map  string // lobby map
+	Name string // server name
+	Rcon *TF2RconWrapper.TF2RconConnection
+
+	League string
+	Type   LobbyType // 9v9 6v6 4v4...
+
+	Address string // server ip:port
+	LobbyId int
+
+	Players        []TF2RconWrapper.Player // current number of players in the server
+	AllowedPlayers []TF2RconWrapper.Player
+
+	Config     *ServerConfig // config that should run before the lobby starts
+	WhiteList  string        // whitelist that should run before the lobby starts
+	MaxPlayers int
+
+	// timer that will verify()
+	Ticker verifyTicker
+
+	//ChatListener  *TF2RconWrapper.RconChatListener
+	RconPassword  string // will store the rcon password specified by the client
+	LobbyPassword string // will store the new server password from the lobby
+}
+
+// timer used in verify()
+type verifyTicker struct {
+	Ticker *time.Ticker
+	Quit   chan struct{}
+}
+
+func (t *verifyTicker) Close() {
+	close(t.Quit)
+}
+
+type Map struct {
+	Name   string
+	Config string
+	League string
+	Mode   string
+}
+
+func NewServer() *Server {
+	return new(Server)
+}
+
+// after create the server var, you should run this
+//
+// things that needs to be specified before run this:
+// -> Map
+// -> Mode
+// -> League
+// -> LobbyId
+// -> Address
+// -> RconPassword
+// -> LobbyPassword
+//
+func (s *Server) Setup() error {
+	fmt.Println("[Server.Setup]: Setting up server -> [" + s.Address + "] from lobby [#" + strconv.Itoa(s.LobbyId) + "]")
+
+	// connect to rcon
+	var err error
+	s.Rcon, err = TF2RconWrapper.NewTF2RconConnection(s.Address, s.RconPassword)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// changing server password
+	passErr := s.ChangePassword(s.LobbyPassword)
+
+	if passErr != nil {
+		log.Fatal(passErr)
+	}
+
+	// kick players
+	fmt.Println("[Server.Setup]: Connected to server, getting players...")
+	kickErr := s.KickAll()
+
+	if kickErr != nil {
+		log.Fatal(kickErr)
+	} else {
+		fmt.Println("[Server.Setup]: Players kicked, running config!")
+	}
+
+	// run config
+	config := NewServerConfig()
+	config.League = s.League
+	config.Type = s.Type
+	config.Map = s.Map
+	cfg, cfgErr := config.Get()
+
+	if cfgErr == nil {
+		config.Data = cfg
+		configErr := s.ExecConfig(config)
+
+		if configErr != nil {
+			log.Fatal(configErr)
+		}
+	} else {
+		log.Fatal(cfgErr)
+	}
+
+	// change map
+	mapErr := s.ChangeMap(s.Map)
```

Comment:
no checking for an invalid map? What if I enter "asdf; rcon rcon_password 'iownthisnow'; rcon say 'h4xed';"
I'd suggest making Map a type alias of string, and giving it a method `Valid()` that would check maps against a list